.github/workflows/StaticTypeCheck.yml

```diff
@@ -44,6 +44,11 @@ on:
         description: 'Commands to run the static type checks.'
         required: true
         type: string
+      allow_failure:
+        description: 'Allow a type checking to be failing without failing the overall workflow.'
+        required: false
+        default: false
```

Comment:
@umarcor this is a breaking change.

I prefer to set it to false as it is the default behavior of any job. OTOH, it's very unlikely to have a mypy run without any error :(.